packages/relay/src/lib/eth.ts

```diff
@@ -1316,24 +1316,44 @@ export class EthImpl implements Eth {
       return [];
     }
 
-    return result.logs.map(log => {
-      return new Log({
-        address: log.address,
-        blockHash: EthImpl.toHash32(log.block_hash),
-        blockNumber: EthImpl.numberTo0x(log.block_number),
-        data: log.data,
-        logIndex: EthImpl.numberTo0x(log.index),
-        removed: false,
-        topics: log.topics,
-        transactionHash: EthImpl.toHash32(log.transaction_hash),
-        transactionIndex: EthImpl.numberTo0x(log.transaction_index)
-      });
-    });
+    const logs: Log[] = [];
+    for(const log of result.logs) {
+      logs.push(
+        new Log({
+          address: await this.getLogEvmAddress(log.address, requestId) || log.address,
+          blockHash: EthImpl.toHash32(log.block_hash),
+          blockNumber: EthImpl.numberTo0x(log.block_number),
+          data: log.data,
+          logIndex: EthImpl.numberTo0x(log.index),
+          removed: false,
+          topics: log.topics,
+          transactionHash: EthImpl.toHash32(log.transaction_hash),
+          transactionIndex: EthImpl.numberTo0x(log.transaction_index)
+        })
+      );
+    }
+
+    return logs;
   }
 
   async maxPriorityFeePerGas(requestId?: string): Promise<string> {
     const requestIdPrefix = formatRequestIdMessage(requestId);
     this.logger.trace(`${requestIdPrefix} maxPriorityFeePerGas()`);
     return EthImpl.zeroHex;
   }
+
+  private async getLogEvmAddress(address: string, requestId: string | undefined) {
+    const cachedLabel = `getLogEvmAddress.${address}`;
+    let contractAddress = cache.get(cachedLabel);
+
+    // If contractAddress === undefined it means there's no cache record
+    // and a mirror node request should be executed.
+    if(contractAddress === undefined) {
+      const contract = await this.mirrorNodeClient.getContract(address, requestId);
+      contractAddress = contract.evm_address;
+      cache.set(cachedLabel, contractAddress, constants.CACHE_TTL.ONE_HOUR);
```

Comment:
It seems `js-cache` doesn't support a fixed size? I hope you don't get a lot of requests in that one hour and cause a memory overflow. This is a security issue. You guys should investigate an alternative library in a follow up.

Reply:
Added https://github.com/hashgraph/hedera-json-rpc-relay/issues/658 for next sprint